### PR TITLE
feat(grid): [grid] support resizable config

### PR DESCRIPTION
**Pull request by gweesin**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1602

## What is the new behavior?

We can use `resizableConfig.limit: ({ field, width }) => number` to handle the min-max of each column width limit

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Introduced resizable columns feature in grid and transfer box components.
  - Added multilingual support for resizable column descriptions.

- **Enhancements**
  - New configuration options to set and limit column width adjustments.
  - Improved drag behavior for resizing columns.

- **Testing**
  - Added test scenarios to validate resizing functionality using Playwright.

- **Documentation**
  - Updated demos with examples of resizable grid configurations.

### CODE REVIEW

File: examples/sites/demos/apis/transfer.js [hunk 1]
```diff
@@ -29,6 +29,18 @@ export default {
...
           pcDemo: 'nested-table'
         },
+        {
+          name: 'resizable-config',
+          typeAnchorName: 'grid#IResizableConfig',
+          type: 'IResizableConfig',
+          defaultValue: '',
+          desc: {
+            'zh-CN': '设置穿梭框的列宽可调整参数',
+            'en-US': 'Set the column width adjustment parameters for the transfer box'
```
Comment (gimmyhehe) [inline examples/sites/demos/apis/transfer.js]: @gweesin 非常感谢您的贡献，仔细review了PR代码，发现此处的transfer并没有`resizable-config` prop

Comment (gweesin) [reply examples/sites/demos/apis/transfer.js]: 非常感谢你的指正，检查了一下代码，这似乎是我的一个错误提交，目前 Transfer 组件并不需要支持这样的属性，文档也与实际不符，我稍后调整一下。

**Comment by kagol** [comment]
@gweesin 麻烦解决下冲突~
image

**Review by gimmyhehe — DISMISSED** [review]

File: examples/sites/demos/apis/grid.js [hunk 2]
```diff
@@ -723,6 +723,19 @@ export default {
           pcDemo: 'grid-size#size-resize-column-width',
           mfDemo: ''
         },
+        {
+          name: 'IResizableConfig',
```
Comment (zzcr) [inline examples/sites/demos/apis/grid.js:727]: 这个属性的名称是不是叫resizableConfig，可以标注下这个属性是什么版本添加的哈，预计应该是3.19.0发布这个特性哈，可以参考：[官网组件以及API版本状态维护信息管理方案](https://github.com/opentiny/tiny-vue/wiki/%E5%AE%98%E7%BD%91%E7%BB%84%E4%BB%B6%E4%BB%A5%E5%8F%8AAPI%E7%89%88%E6%9C%AC%E7%8A%B6%E6%80%81%E7%BB%B4%E6%8A%A4%E4%BF%A1%E6%81%AF%E7%AE%A1%E7%90%86%E6%96%B9%E6%A1%88)

Comment (gweesin) [reply examples/sites/demos/apis/grid.js:727]: 了解了，顺便替一下 wiki 需要更新：
image

1. meta -> metaData
2. 多了一种 new 的状态

**Review by zzcr — APPROVED** [review]

**Pull request merged**